### ISSUE TITLE
DocLinkChecker Doesn't understand File Inclusion links

### ISSUE THREAD

**Issue by AMSDavis** [Ellerbach/docfx-companion-tools#29]
When running DocLinkChecker it appears that the checker doesn't understand or find the links if I am using a [file inclusion with a Code Snippet](https://vicancy.github.io/docfx/spec/docfx_flavored_markdown.html#file-inclusion).

These links are rendered just find by docfx however DocLinkChecker believes they are an error.

As an example, if I have the following structure:
```
doc/
    index.md
    docfx.json
    scripts/
        foo.cs
```

and foo.cs has the following content like this:
```
// (other content above)

// <Bravo>
foreach(var thing in list)
{
     // Do Stuff
}
// </Bravo>
```

I can include the entire file with:
```markdown
[!code-csharp[Code](~/Scripts/foo.cs)]
```

But this will be detected as an error by DocLinkChecker:
```markdown
[!code-csharp[Code](~/Scripts/foo.cs?name=Bravo)]
```

**Comment by Ellerbach** [comment]
This should be fixed in a coming release. See https://github.com/Ellerbach/docfx-companion-tools/tree/doclinkchecker-upgrade
You can build and test it to check if it now works better. Please report back if possible.

**Comment by Ellerbach** [comment]
@AMSDavis did you have a chance to test the new version published? It should fix your issue! Let us know so we can close this issue.

**Comment by Pixelrella** [comment]
Just got the same issue in version 1.15.0.

I get warnings for line and region inclusions of csharp code snippets:

```markdown
[!code-csharp[](../DemoProject/MyFile.cs#starting)]
```
```csharp
private void MyMethod()
{
   #region starting
   // Some code
   #endregion
}
```

`***WARNING Heading 'starting' not found in '../DemoProject/MyFile.cs'`

Maybe this should be reported as separate issue, but I am also getting warnings for [tabs](https://dotnet.github.io/docfx/docs/markdown.html?tabs=linux%2Cdotnet#tabs):

```markdown
## [Guide](#tab/text)
```
`***WARNING Heading 'tab/text' not found in '.\Tabbed.md`

Thank you!

**Comment by mtirionMSFT** [comment]
@Pixelrella  sorry about that. I haven't had time to test that scenario. I have added this to the backlog.

**Comment by mtirionMSFT** [comment]
@Pixelrella I've been investigating this problem and it's harder then it looks. The functionality you are describing is not standard markdown (supported by markdig), but it's special for DocFx. There use to be a special markdig extension for these DocFx specials, but that's not supported anymore. I'm trying to see if there is a way to 'easily' add this support, otherwise I might end up writing our own markdig extension to solve this. But this will probably take some time. 

So, just to let you know we're working on it :)

**Comment by mtirionMSFT** [comment]
@AMSDavis @Pixelrella Sometimes it's good to drop things for a while and then come back to it ... to find that it is really simple to solve. I have solved this issue in PR #58.
